### PR TITLE
Fix lazy loading

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️

### CODE REVIEW

**Comment by samdark** [comment]
Something interesting is going on with tests...

**Review by samdark — APPROVED** [review]

File: src/Gii.php [hunk 1]
```diff
@@ -13,31 +12,24 @@
...
      */
-    public function __construct(private array $generators)
+    public function __construct(private array $proxies)
     {
     }
 
     public function addGenerator(GeneratorInterface $generator): void
     {
-        $this->generators[$generator::getId()] = $generator;
+        $this->proxies[$generator::getId()] = new GeneratorProxy(fn () => $generator, $generator::class);
```
Comment (vjik) [inline src/Gii.php:23]: Method gets ready-to-use generator. Seems, proxy not need here.
Or method should allow to pass array configuration of generator.

File: src/Gii.php [hunk 2]
```diff
@@ -13,31 +12,24 @@
...
     }
 
     public function getGenerator(string $id): GeneratorInterface
     {
-        if (!isset($this->generators[$id])) {
-            throw new GeneratorNotFoundException('Generator "' . $id . '" not found');
-        }
-
-        return $this->generators[$id] instanceof Closure ? $this->generators[$id]() : $this->generators[$id];
+        return $this->proxies[$id]?->loadGenerator() ?? throw new GeneratorNotFoundException('Generator "' . $id . '" not found');
```
Comment (vjik) [inline src/Gii.php:28]: Proxies can contain ready-to-use generators (`array<string, GeneratorInterface|GeneratorProxy> $proxies`).

Comment (xepozz) [reply src/Gii.php]: Done

File: src/Gii.php (hunk 1 above)
Comment (xepozz) [reply src/Gii.php]: Done

**Review by samdark — APPROVED** [review]

File: src/GeneratorProxy.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Gii;
+
+use Closure;
+
+final class GeneratorProxy
+{
+    private ?GeneratorInterface $generator = null;
```
Comment (vjik) [inline src/GeneratorProxy.php:11]: Seems, this property never used and always "null".

Comment (xepozz) [reply src/GeneratorProxy.php:11]: It's used for memoization

**Pull request merged**